components/brave_wallet/renderer/brave_wallet_js_handler.cc

```diff
@@ -91,11 +93,23 @@ void BraveWalletJSHandler::BindFunctionToObject(
       .Check();
 }
 
-v8::Local<v8::Promise> BraveWalletJSHandler::Request(v8::Isolate* isolate,
-                                                     const std::string& input) {
-  if (!EnsureConnected())
+v8::Local<v8::Promise> BraveWalletJSHandler::Request(
+    v8::Isolate* isolate,
+    v8::Local<v8::Value> input) {
+  if (!EnsureConnected() || !input->IsObject())
+    return v8::Local<v8::Promise>();
+
+  std::unique_ptr<base::Value> out(
+      content::V8ValueConverter::Create()->FromV8Value(
+          input, isolate->GetCurrentContext()));
+
+  base::DictionaryValue* out_dict;
+  if (!out || !out->is_dict() || !out->GetAsDictionary(&out_dict))
     return v8::Local<v8::Promise>();
 
+  std::string formed_input;
+  base::JSONWriter::Write(*out_dict, &formed_input);
```

Comment:
shouldn't we also check the return value?

Reply:
good point, I've added it